src/systems/systems/src/hydraulic/update_iterator.rs

```diff
@@ -0,0 +1,292 @@
+use crate::simulation::UpdateContext;
+use std::{iter::Iterator, time::Duration};
+
+pub struct FixedStepLoop {
+    lag_time_accumulator: Duration,
+    time_step: Duration,
+    number_of_loops_this_iteration: u32,
+    number_of_loops_remaining: u32,
+}
+impl FixedStepLoop {
+    pub fn new(time_step: Duration) -> Self {
+        Self {
+            lag_time_accumulator: Duration::from_millis(0),
+            time_step,
+            number_of_loops_this_iteration: 0,
+            number_of_loops_remaining: 0,
+        }
+    }
+
+    fn time_step(&self) -> Duration {
+        // println!("fixed::time_step: time{:.3}", self.time_step.as_secs_f64(),);
+        self.time_step
+    }
+
+    fn _loop_number(&self) -> u32 {
+        self.number_of_loops_this_iteration
+    }
```

Comment:
With `_` because it is unused? If so, then please remove, otherwise rename to not have `_`.